content/blog/nats-server-2.9.17-release.md

```diff
@@ -0,0 +1,124 @@
++++
+date = "2023-05-19"
+draft = false
+title = "NATS Server 2.9.17 Release"
+author = "Byron Ruth"
+categories = ["Engineering"]
+tags = ["NATS", "Release"]
++++
+
+The NATS maintainers are proud to announce the [2.9.17 release](https://github.com/nats-io/nats-server/releases/tag/v2.9.17)! We want to thank all of the people who contributed to this release! If **you** are interested in contributing, please check out [all the ways you can](https://nats.io/contributing/)!
+
+The key areas this post will cover include:
+
+- [WebSocket transport truncation](#-websocket-transport-truncation)
+- [Health monitor improvements](#-health-monitor-improvements)
+- [Leafnode fleet optimizations](#-leafnode-fleet-optimizations)
+- [Monotonic time calculations](#-monotonic-time-calculations)
+
+For the entirety of the improvements and fixes, check out the [release notes](https://github.com/nats-io/nats-server/releases/tag/v2.9.17).
+
+## 🐘 WebSocket transport truncation
+
+As part of the [P99 peformance work in the 2.9.16 release](https://nats.io/blog/nats-server-2.9.16-release/#-p99-performance), a regression was introduced, resulting in [faulty subscription behavior in clients](https://github.com/nats-io/nats.ws/issues/194) received fairly large messages over the WebSocket transport. Internally this had to do with prematurely deallocating a buffer to a shared pool before data was entirely flushed on a connection.
+
+However, the upside of addressing this issue resulted in more (correct) optimizations for the WebSocket transport, including reducing the overall memory footprint and switching to a [faster compression algorithm](https://github.com/nats-io/nats-server/pull/4087).
+
+**Relevant PRs**
+
+- [#4084](https://github.com/nats-io/nats-server/pull/4084)
+- [#4087](https://github.com/nats-io/nats-server/pull/4087)
+- [#4093](https://github.com/nats-io/nats-server/pull/4093)
+
+## ⚕️ Health monitor improvements
+
+NATS provides a set of [monitoring endpoints](https://docs.nats.io/running-a-nats-service/nats_admin/monitoring) containing a variety of information about the current state of the server.
+
+The job of the `/healthz` endpoint is to return `200 OK` when the server is ready for client connections and, if JetStream is enabled, all of the assets (streams and consumers) on the server have been accounted for, any necessary recovery was performed, and the integrity of the data has been verified.
+
+In rare situations, for clusters with a large number of assets, or if the asset sizes are large, it was possible for given server to get stuck and never finish the internal checks, therefore never becoming ready for client connections. A forced server restart would typically fix this issue, however, clearly this was not ideal.
```

Comment:
"for given server" ... might use "for a given server" but probably best to just be "for a server"